### PR TITLE
=doc amending mapAsyncUnordered docs to include parallelism (#21556)

### PR DESCRIPTION
**Pull request by svezfaz**
Refs #21556

### CODE REVIEW

**Review by ktoso — APPROVED** [review]
Good changes, thanks!

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 1]
```diff
@@ -1811,8 +1812,8 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
   def detach: javadsl.Flow[In, Out, Mat] = new Flow(delegate.detach)
 
   /**
-   * Materializes to `Future[Done]` that completes on getting termination message.
-   * The Future completes with success when received complete message from upstream or cancel
+   * Materializes to `CompletionStage[Done]` that completes on getting termination message.
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Should be `CompletionStage<Done>` even? (If you don't mind boyscouting a bit more :) )

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala (hunk 1 above)
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Great, thanks!

**Pull request merged**